core/src/main/java/hudson/Functions.java

```diff
@@ -776,10 +796,33 @@ public static String urlEncode(String s) {
         return URLEncoder.encode(s, StandardCharsets.UTF_8);
     }
 
+    /**
+     * Escapes HTML unsafe characters
+     * <pre>
+     * Input example  1: &amp; " ' &lt; &gt;
+     * Output example 1: &amp; &quot; &#039; &lt; &gt;
+     * Input example  2: !"£$%^&amp;*()_+}{:@~?&gt;&lt;|¬`,./;'#[]-=
+     * Output example 2: !&quot;£$%^&amp;*()_+}{:@~?&gt;&lt;|¬`,./;&#039;#[]-=&nbsp;
```

Comment:
Outputs need to be double-escaped, otherwise they're going to be rendered the same as input in HTML Javadoc. From IntelliJ preview:

<img width="598" alt="Screenshot" src="https://user-images.githubusercontent.com/1831569/215861445-716b5900-0731-4232-92a5-be449c42733c.png">